otelx/otel.go

```diff
@@ -0,0 +1,65 @@
+package otelx
+
+import (
+	"github.com/ory/x/logrusx"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Tracer struct {
+	Config *Config
+
+	l      *logrusx.Logger
+	tracer trace.Tracer
+}
+
+// Creates a new tracer. If name is empty, a default tracer name is used
+// instead. See: https://godocs.io/go.opentelemetry.io/otel/sdk/trace#TracerProvider.Tracer
+func New(name string, l *logrusx.Logger, c *Config) (*Tracer, error) {
+	t := &Tracer{Config: c, l: l}
+
+	if err := t.setup(name); err != nil {
+		return nil, err
+	}
+
+	return t, nil
+}
+
+// Creates a new no-op tracer.
+func NewNoop(l *logrusx.Logger, c *Config) *Tracer {
+	tp := trace.NewNoopTracerProvider()
+	t := &Tracer{Config: c, l: l, tracer: tp.Tracer("")}
+	return t
+}
+
+// setup sets up the tracer.
+func (t *Tracer) setup(name string) error {
+	switch t.Config.Provider {
```

Comment:
You can use https://github.com/ory/x/blob/e49f326710518506477a020f5d7eeef02f92de5c/stringsx/switch_case_test.go#L59-L66
That will automatically create a nice error with all valid cases.

Reply:
Ok I've added this. I hope I've used it right. :)